File: Problem3/code.py
def factors_of(n):
    factors = set()
    div = 2
    
    while n >= 2:
        if n % div == 0:
            factors.add(div)
            
            n = n / div
        else:
            div += 1

    return factors.pop()

print(factors_of(600851475143)) #6857

File: Problem 3/code.js
function factorsOf(n) {
    let factors = [];
    let div = 2;
  
    while (n >= 2) {
      if (n % div == 0) {
        factors.push(div);
        n = n / div;
      } else {
        div++;
      }
    }


    return factors[factors.length-1];
  }

console.log(factorsOf(600851475143)) // 6857


